### PR TITLE
profiles: fix index in example

### PR DESCRIPTION
**Pull request by florianl**
For `abc;def` the `locations_start_index` should be `4` as `2` points to `baz`.

Follow up of https://github.com/open-telemetry/oteps/pull/239#discussion_r1496062414

### CODE REVIEW

**Review by joaopgrassi — APPROVED** [review]

File: text/profiles/0239-profiles-data-model.md [hunk 1]
```diff
@@ -1349,7 +1349,7 @@ sample:
     locations_length: 3
     value:
       - 100
-  - locations_start_index: 2
+  - locations_start_index: 4
```
Comment (petethepig) [inline text/profiles/0239-profiles-data-model.md:1352]: Ah, my bad, thanks for fixing this, but I think it should be `3` actually:
```suggestion
  - locations_start_index: 3
```

And also, I think `location_indices` should be slightly different, there's a off by 1 error right now:
```
location_indices:
  - 0 # foo
  - 1 # bar
  - 2 # baz
  - 3 # abc
  - 4 # def
```

@florianl Is this right or am I missing something?

**Review by petethepig — CHANGES_REQUESTED** [review]

**Comment by mtwo** [comment]
Comment from the OTel maintainer meeting: could / should this be moved to a comment on the current [Profiling PR in the OTLP](https://github.com/open-telemetry/opentelemetry-proto/pull/534) repository?

File: text/profiles/0239-profiles-data-model.md (hunk 1 above)
Comment (florianl) [reply text/profiles/0239-profiles-data-model.md:1352]: sorry for the delay - you are right :sweat_smile:

File: text/profiles/0239-profiles-data-model.md [hunk 2]
```diff
@@ -1349,7 +1349,7 @@ sample:
     locations_length: 3
     value:
       - 100
-  - locations_start_index: 2
+  - locations_start_index: 3
```
Comment (trask) [inline text/profiles/0239-profiles-data-model.md:1352]: also, should below

```
location_indices:
  - 0 # foo
  - 1 # bar
  - 2 # baz
  - 4 # abc
  - 5 # def
```

be

```
location_indices:
  - 0 # foo
  - 1 # bar
  - 2 # baz
  - 3 # abc
  - 4 # def
```

**Review by trask — APPROVED** [review]
I think it's worth fixing here since it's just editorial and can cause confusion for readers.

File: text/profiles/0239-profiles-data-model.md (hunk 2 above)
Comment (florianl) [reply text/profiles/0239-profiles-data-model.md:1352]: Good spot! :+1: I have fixed also the indices of `location_indices`.

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**